pallets/gear/src/manager.rs

```diff
@@ -101,6 +133,27 @@ impl<T: Config> ExtManager<T>
 where
     T::AccountId: Origin,
 {
+    /// Check if id is program.
+    pub fn is_program(&mut self, id: &ProgramId) -> bool {
+        // TODO: research how much need to charge for `program_exists` query.
+        if self.programs.contains(id) {
+            true
+        } else if self.users.contains(id) {
+            false
+        } else if GearProgramPallet::<T>::program_exists(*id) {
+            self.programs.insert(*id);
+            true
+        } else {
+            self.users.insert(*id);
+            false
+        }
+    }
+
+    /// Check if id is not program.
+    pub fn is_user(&mut self, id: &ProgramId) -> bool {
```

Comment:
Frankly speaking this signature and `is_program` are looking weird

Reply:
any suggestions?

Reply:
Maybe just rename to something like `check_if_user_and_add`/`if_user_remember`? Usually functions named `is_something` don't have to accept `&mut self`. Anyway these two are used there only so remark is not important